torch/csrc/distributed/c10d/TCPStoreLibUvBackend.cpp

```diff
@@ -134,6 +137,9 @@ class UvTcpSocket : public UvHandle {
  public:
   explicit UvTcpSocket(uv_loop_t* loop) {
     uv_tcp_init(loop, &client);
+    if(int err = uv_tcp_nodelay(&client, 1)) {
```

Comment:
do we have some microbenchmark code that we could use to validate the improvement (and no unexpected perf impact)?

Reply:
Yes, I'll do some more testing on this internally but I expect it'll be pretty safe.

We already set TCP_NODELAY on everything that uses socket.cpp so the older TCPStoreBackend and the current TCPStore client. Not setting this seems like an oversight when writing the new backend -- it's a bit weird to set it only on the client but not the server as well since it's asymmetric.

https://github.com/pytorch/pytorch/blob/4b96575a09b167489539d8916e86cf6c13e9544d/torch/csrc/distributed/c10d/socket.cpp#L358

Reply:
One more data point: we do internal buffering of messages so we only call write() once the entire message is ready to be sent. Thus we only call write once per request so there won't ever be another message that can be coalesced.

https://github.com/pytorch/pytorch/blob/0975d5ed6e63a68fbca3ae55b8e3785a557fb6ac/torch/csrc/distributed/c10d/TCPStoreLibUvBackend.cpp#L444